sdkproject/Assets/Mapbox/Unity/MeshGeneration/Modifiers/MeshModifiers/HeightModifier.cs

```diff
@@ -79,18 +79,19 @@ public override void Run(VectorFeatureUnity feature, MeshData md, UnityTile tile
                 {
                     md.Vertices[i] = new Vector3(md.Vertices[i].x, md.Vertices[i].y + minHeight + hf, md.Vertices[i].z);
                 }
-            }           
+            }
 
-            var count = md.Vertices.Count;
-            float d = 0f;
-            Vector3 v1;
-            Vector3 v2;
-            int ind = 0;
+			var count = md.Vertices.Count;
+			md.Vertices.Capacity = count + md.Edges.Count * 2;
+			float d = 0f;
+			Vector3 v1;
+			Vector3 v2;
+			int ind = 0;
 
-            var wallTri = new List<int>();
-            var wallUv = new List<Vector2>();
+			var wallTri = new List<int>(md.Edges.Count * 3);
```

Comment:
If we know the size, we should probably just use arrays. In fact, one future optimization might be to pool arrays in some memory manager singleton (referenced as `Scratch Arrays` here: https://youtu.be/F6vtgGY-aLs?t=31m20s).